cmd/restic/cmd_check.go

```diff
@@ -271,6 +282,8 @@ func runCheck(ctx context.Context, opts CheckOptions, gopts GlobalOptions, args
 			printer.E("error: %v\n", err)
 		}
 
+		result.NumErrors += len(errs)
+		result.HintRepairIndex = true
 		printer.E("\nThe repository index is damaged and must be repaired. You must run `restic repair index' to correct this.\n\n")
 		return errors.Fatal("repository contains errors")
```

Comment:
`result` is never printed  in this case.

Reply:
The summary wasn't printed in any of the early returns. I added 1a2b9ba37ed0309d42b2cede8836fe99cf9f9d56 to always print the summary JSON.